src/spox/_attributes.py

```diff
@@ -210,9 +210,16 @@ def maybe(
         return cls(tuple(value), name) if value is not None else None
 
     def _to_onnx_deref(self) -> AttributeProto:
-        return make_attribute(
-            self._name, self.value, attr_type=self._attribute_proto_type
-        )
+        # the attr_type argument was only added in onnx>=1.15. It is a
+        # performance optimization for which we don't want to
+        # introduce a version bound. If we fail, we try again without
+        # it.
+        try:
```

Comment:
Could we explicitly check the ONNX version instead of relying on catching TypeError? TypeErrors could be raised for [different](https://github.com/onnx/onnx/blob/48501e0ff59843a33f89e028cbcbf080edd0c1df/onnx/helper.py#L972) reasons which should surface here.